### ISSUE TITLE
Sent message progress is infinite.

### ISSUE THREAD

**Issue by elevenetc** [GetStream/stream-chat-android#247]
Sometimes is hard to reproduce.

**Comment by elevenetc** [comment]
Debugged the issues. It's race condition problem and not API/Socket problem. The method of `Message` is called in several places:
```java
public void setStatus(MessageStatus status) {
    Log.d(TAG, "setStatus: " +  getText() + ": " + status);
    this.status = status;
}
```
And by sending one message it is called 5 times:
```java
D  setStatus: null: null
D  setStatus: test: SENDING
D  setStatus: test: SENDING
D  setStatus: test: RECEIVED
D  setStatus: test: SENDING
D  setStatus: test: RECEIVED
```
Somewhere in between `MessageListItemViewHolder` updates view and sees `SENDING` status and sets infinite progress.

**Comment by adrian09h** [comment]
@elevenetc  this was fixed from PR #161. waiting for review and merge
